raiden/network/rpc/client.py

```diff
@@ -631,7 +631,10 @@ def __init__(self, jsonrpc_client, registry_address, startgas=GAS_LIMIT,
 
     def manager_address_by_token(self, token_address):
         """ Return the channel manager address for the given token. """
-        return self.proxy.channelManagerByToken.call(token_address)
+        manager_address_encoded = self.proxy.channelManagerByToken.call(token_address)
+        if not manager_address_encoded or manager_address_encoded == '0x':
+            manager_address_encoded = self.add_token(token_address)
```

Comment:
You suggest we add one more function on top of it which would also register it? Or since it's only used in two places and presumably in both places we would need it registered if it does not exists simply rename it to something like: `maybe_register_token_and_get_manager()`